sdk/eventhub/azure-eventhub/azure/eventhub/_client_base.py

```diff
@@ -229,7 +229,12 @@ def _management_request(self, mgmt_msg, op_type):
                     status_code_field=b"status-code",
                     description_fields=b"status-description",
                 )
-                return response
+                status_code = response.application_properties[b"status-code"]
+                if status_code < 400:
+                    return response
+                raise errors.AuthenticationException(
```

Comment:
Are we sure `AuthenticationException` is the right error here?
What if this happened during the `list_partitions` mgmt request?

Reply:
TBH, I'm not sure about it, it's empirical. but I'm okay with escalating it to more general error like `EventHubError`.

Reply:
as for the `list_partitions`? do you mean the `list_partitions` in ownership manager when an eventprocessor tries to claim ownership?

The load-balancing algorithm will take this as normal error, catch it, do logging and then keep retrying.

Reply:
Currently every mgmt call creates a new connection. I think it makes sense to raise an authentication exception if it fails.
AuthenticationException is an uamqp error. uamqp doesn't raise an exception but returns empty data and wrong status code. So this is like workaround that re-defines uamqp behavior here. The try ... except... here will catch the exception and retry.
If it fails after all retry, an event hub AuthenticationError is raised to the users.

Reply:
list_partitions is the same mgmt request. So it will get an event hub AuthenticationError